### ISSUE TITLE
Chapter duration is not detetected correctly

### ISSUE THREAD

**Issue by yermak** [yermak/AudioBookConverter#21]
Most of the tools and api can give just approximate duration of source mp3 file based on its bitrate, thus final duration of chapter might be incorrect.
Probably duration of the chapter might need to detected based on final outcoming m4b file.

**Comment by jayrodathome** [comment]
FYI Love this program so much. Reminds me of Audiobookbuilder on Mac. Anyway I noticed the chapters cut kinda short but not always at the end. Maybe have an option to just add a couple second delay between chapters. Not sure what kind of work that would entail I'm not a coder. Just seems audible files have a small pause between chapters and having that as an option when building the book would be nice. Cheers for all of your hard work! Much appreciated. Can we donate?

**Comment by yermak** [comment]
Good idea to add 2sec of delay. Will work on it.

**Comment by sandreas** [comment]
Are you detection the chapter length with `ffmpeg`? If this is the case, let me tell you that `ffmpeg` is inaccurate detecting the chapter length... `mp4v2` is the only tool i came across that was able to detect the lengths of `mp4` files correctly (`mp4info` is the tool that worked for me)

**Comment by yermak** [comment]
Sandreas, thank you very much for the hint. I also noticed that. FFmpeg even for mp3 files detects duration based on bitrate - not sure how accurate it is with VBR, I tried to use it on resulting mp4 - not sure if that gave any noticible accuracy. 
I will try to restore my enviroment for mp4v2 and rebuild mp4info.

**Comment by sandreas** [comment]
Here are some tests, that i did for length detection in `m4b-tool` using a file `testing.m4b`:

**full stream analysis (takes pretty long on big files and is inaccurate)**
**Result: 00:00:22.12 (22120ms)**
```
ffmpeg -hide_banner -i testing.m4b -f null -

...
frame=    1 fps=0.0 q=-0.0 Lsize=N/A time=00:00:22.12 bitrate=N/A speed= 350x    
video:1kB audio:3812kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: unknown
...
```


**metadata analysis (pretty fast but not 100% accurate, maybe only due to rounding errors)**
**Result: 00:00:22.15 (22150ms)**
```
ffmpeg -hide_banner -i testing.m4b -f ffmetadata -
...
 Duration: 00:00:22.15
...
```

**mp4info metadata analysis (fast AND 100% accurate)**
**Result: 22.149 secs (22149ms)**
```
mp4info testing.m4b

...
audio   MPEG-4 AAC LC, 22.149 secs, 127 kbps, 44100 Hz
...
```


**Conclusion** 
Neither `ffmpeg` nor `ffprobe` are accurate enough for very long audio streams. Since I did not find a better way, I use `mp4info`.

**Comment by yermak** [comment]
This issue should be fixed in 4.1.0 by replacing detection of chapter duration from ffmpeg on source mp3 with mp4v2 on output m4b.
@sandreas, thank you for insipiration. 
@jayrodathome, i hope 2 second break between chapters is not needed.
